client/trino-client/src/main/java/io/trino/client/auth/external/LocalKnownToken.java

```diff
@@ -24,7 +22,7 @@
  * LocalKnownToken class keeps the token on its field
  * and it's designed to use it in fully serialized manner.
  */
-@NotThreadSafe
```

Comment:
we have 42 cases like this. i think we should replace it with our own annotation (can be prep commit / separate PR)

Reply:
Where should I add this annotation to? Spi?

Reply:
let's add it to plugin toolkit for now. not awesome place, but this shouldn't be in SPI

the only two usages in spi and client will stll need to be converted to a code comment.

Reply:
But what's the overall meaning of this annotation? If not stated otherwise (`@ThreadSafe`) I'd assume that _every_ class is not thread-safe (the default). Do we even need to state it? It's like `@Immutable` vs `@Mutable` (the default)?

Reply:
I'm fine with the introduction of our own annotation but I'd like to understand the meaning and value of it. Maybe we should state that it's dangerous to use the class without synchronization. i.e. `@DangerouslyNotThreadSafe`

Reply:
`@NotThreadSafe` is enough.

Reply:
> But what's the overall meaning of this annotation? If not stated otherwise (`@ThreadSafe`) I'd assume that _every_ class is not thread-safe (the default).

yes, but the annotation documents the intent.
also, some classes that must be thread safe are not marked as `@ThreadSafe`, to `@NotThreadSafe` saves the reader from having to guess.